igvc_perception/src/pointcloud_segmentation/utils.h

```diff
@@ -0,0 +1,168 @@
+  msg.lifetime = ros::Duration(0.1);
+  msg.id = id;
+  msg.action = visualization_msgs::Marker::ADD;
+  msg.color.r = 1.0f;
+  msg.color.g = 0.0f;
+  msg.color.b = 0.0f;
+  msg.color.a = 1.0f;
+  msg.scale.x = 0.05;
+  msg.pose.orientation.w = 1.0;
+  msg.points.clear();
+
+  geometry_msgs::Point p1, p2, p3, p4, p5, p6, p7, p8;
+  p1.x = p2.x = p5.x = p6.x = cloud.min[0];
+  p3.x = p4.x = p7.x = p8.x = cloud.max[0];
+  p1.y = p3.y = p5.y = p7.y = cloud.min[1];
+  p2.y = p4.y = p6.y = p8.y = cloud.max[1];
+  p1.z = p2.z = p3.z = p4.z = cloud.min[2];
+  p5.z = p6.z = p7.z = p8.z = cloud.max[2];
+  msg.points.push_back(p1);
+  msg.points.push_back(p2);
+  msg.points.push_back(p1);
+  msg.points.push_back(p3);
+  msg.points.push_back(p2);
+  msg.points.push_back(p4);
+  msg.points.push_back(p3);
+  msg.points.push_back(p4);
+  msg.points.push_back(p1);
+  msg.points.push_back(p5);
+  msg.points.push_back(p2);
+  msg.points.push_back(p6);
+  msg.points.push_back(p3);
+  msg.points.push_back(p7);
+  msg.points.push_back(p4);
+  msg.points.push_back(p8);
+  msg.points.push_back(p5);
+  msg.points.push_back(p6);
+  msg.points.push_back(p5);
+  msg.points.push_back(p7);
+  msg.points.push_back(p6);
+  msg.points.push_back(p8);
+  msg.points.push_back(p7);
+  msg.points.push_back(p8);
+
+  return msg;
+};
+
+std::vector<pcl::PointIndices> euclidean_clustering(PC::Ptr input_cloud)
+{
+  pcl::search::KdTree<pcl::PointXYZ>::Ptr tree(new pcl::search::KdTree<pcl::PointXYZ>);
+  tree->setInputCloud(input_cloud);
+  std::vector<pcl::PointIndices> cluster_indices;
+  pcl::EuclideanClusterExtraction<pcl::PointXYZ> ec;
+
+  ec.setClusterTolerance(0.25);
+  ec.setMinClusterSize(20);
+  ec.setMaxClusterSize(300);
+  ec.setSearchMethod(tree);
+  ec.setInputCloud(input_cloud);
+  ec.extract(cluster_indices);
+  return cluster_indices;
+};
+
+std::vector<pcl::PointIndices> region_growing_clustering(PC::Ptr input_cloud)
+{
+  pcl::search::KdTree<pcl::PointXYZ>::Ptr tree(new pcl::search::KdTree<pcl::PointXYZ>);
+  tree->setInputCloud(input_cloud);
+  pcl::PointCloud<pcl::Normal>::Ptr normals(new pcl::PointCloud<pcl::Normal>);
+  pcl::NormalEstimation<pcl::PointXYZ, pcl::Normal> normal_estimator;
+  normal_estimator.setSearchMethod(tree);
+  normal_estimator.setInputCloud(input_cloud);
+  normal_estimator.setKSearch(30);
+  normal_estimator.compute(*normals);
+
+  pcl::RegionGrowing<pcl::PointXYZ, pcl::Normal> reg;
+  reg.setMinClusterSize(20);
+  reg.setMaxClusterSize(300);
+  reg.setSearchMethod(tree);
+  reg.setNumberOfNeighbours(30);
+  reg.setInputCloud(input_cloud);
+  reg.setInputNormals(normals);
+  reg.setSmoothnessThreshold(3.0 / 180.0 * M_PI);
+  reg.setCurvatureThreshold(1.0);
+
+  std::vector<pcl::PointIndices> clusters;
+  reg.extract(clusters);
+
+  return clusters;
+};
+
+void remove_outlier(PC::Ptr input_cloud)
+{
+  pcl::StatisticalOutlierRemoval<pcl::PointXYZ> sor;
+  sor.setInputCloud(input_cloud);
+  sor.setMeanK(50);
```

Comment:
again, this should be an input and the value should be set from a configuration file.